### PR TITLE
Use cookie as default storage for cross subdomains support

### PR DESCRIPTION
**Pull request by mcarpentierCoveo**
Support cross-domain in PageViewRecommendation script

internal: https://coveord.atlassian.net/browse/JSUI-769

### CODE REVIEW

File: src/cookieutils.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// Code originally taken from : https://developers.livechatinc.com/blog/setting-cookies-to-subdomains-in-javascript/
+export class Cookie {
+  private static prefix: string = 'coveo_';
```
Comment (pastjean) [inline src/cookieutils.ts]: i'd maybe put a different prefix so it doesn't clash with our other cookies `coveo_analytics_js` maybe. Maybe the version in it too or a timestamp/guid

**Comment by pastjean** [comment]
Really nice , with tests of things i didn't test.

Try not to reference our internal issue tracker in open source projects please so everyone has access.

File: src/cookieutils.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    if (expiration) {
+      date = new Date();
+      date.setTime(date.getTime() + expiration);
+      expires = '; expires=' + date.toGMTString();
+    } else {
+      expires = '';
+    }
+
+    host = location.hostname;
+    if (host.split('.').length === 1) {
```
Comment (GermainBergeron) [inline src/cookieutils.ts]: I think `if (host.indexOf('.') === -1)` is easier to read

File: src/cookieutils.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  static get(name: string) {
+    var nameEQ = this.prefix + name + '=';
+    var ca = document.cookie.split(';');
+    for (var i = 0; i < ca.length; i++) {
+      var c = ca[i];
+      while (c.charAt(0) == ' ') {
+        c = c.substring(1, c.length);
+      }
```
Comment (GermainBergeron) [inline src/cookieutils.ts]: Try to use more descriptive variable name, what is `c` ? and `ca`?

File: src/cookieutils.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      if (Cookie.get(name) == null || Cookie.get(name) != value) {
+        // append "." to current domain
+        domain = '.' + host;
+        document.cookie = this.prefix + name + '=' + value + expires + '; path=/; domain=' + domain;
+      }
+    }
+  }
+
+  static get(name: string) {
+    var nameEQ = this.prefix + name + '=';
```
Comment (GermainBergeron) [inline src/cookieutils.ts]: I would rename `nameEQ` to `cookiePrefix` or something similar

File: src/cookieutils.ts (hunk 3 above)
Comment (GermainBergeron) [inline src/cookieutils.ts]: You can strip the whitespace at the start with a regex: `str.replace(/^\s+/, '')`

**Comment by GermainBergeron** [comment]
Good job!

File: src/cookieutils.ts (hunk 1 above)
Comment (mcarpentierCoveo) [inline src/cookieutils.ts]: i removed the prefix to only use the store key ( __coveo.analytics.history ) as the cookie name.

**Comment by pastjean** [comment]
You can merge then bump minor version + tag minor version?

**Pull request merged**